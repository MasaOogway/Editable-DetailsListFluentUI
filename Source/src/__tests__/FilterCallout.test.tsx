import { render } from '@testing-library/react';
import React from 'react';
import FilterCallout from '../libs/editablegrid/columnfiltercallout/filtercallout';
import { vi } from 'vitest';

describe('FilterCallout', () => {
  it('should render Callout component with correct props', () => {
    const columnClass = 'example-column';
    const columnKey = 'example-key';
    const columnName = 'Example Column';
    const filterList = [
      { text: 'Option 1', isChecked: false },
      { text: 'Option 2', isChecked: true },
      { text: 'Option 3', isChecked: false }
    ];

    const onCancel = vi.fn();
    const onApply = vi.fn();

    const { container } = render(
      <FilterCallout
        columnClass={columnClass}
        columnKey={columnKey}
        columnName={columnName}
        filterList={filterList}
        onCancel={onCancel}
        onApply={onApply}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
