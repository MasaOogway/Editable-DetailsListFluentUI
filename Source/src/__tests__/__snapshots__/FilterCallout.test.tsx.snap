// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterCallout > should render Callout component with correct props 1`] = `
<div>
  <span
    class="ms-layer"
  />
</div>
`;
